### PR TITLE
ER-1337 & ER-1344 training confirm

### PR DESCRIPTION
**Pull request by chrisjensenuk**
Added confirmation screens when selecting training for Employer.Web and Provider.Web

### CODE REVIEW

File: src/Provider/Provider.Web/Controllers/Part1/TrainingController.cs [hunk 1]
```diff
@@ -48,12 +106,22 @@ public async Task<IActionResult> Training(TrainingEditModel m, [FromQuery] bool
             {
                 var vm = await _orchestrator.GetTrainingViewModelAsync(m, user);
                 vm.PageInfo.SetWizard(wizard);
+
```
Comment (mrpmorris) [inline src/Provider/Provider.Web/Controllers/Part1/TrainingController.cs]: de-duplicate the VM code, as per previous suggestion

File: src/Employer/Employer.Web/Controllers/Part1/TrainingController.cs [hunk 2]
```diff
@@ -13,52 +16,61 @@ namespace Esfa.Recruit.Employer.Web.Controllers.Part1
...
         public TrainingController(TrainingOrchestrator orchestrator)
         {
             _orchestrator = orchestrator;
         }
         
         [HttpGet("training", Name = RouteNames.Training_Get)]
-        public async Task<IActionResult> Training(VacancyRouteModel vrm, [FromQuery] string wizard = "true", [FromQuery] string clear = "", [FromQuery] string hasTraining = "")
+        public async Task<IActionResult> Training(VacancyRouteModel vrm, [FromQuery] string wizard = "true", [FromQuery] string clear = "", [FromQuery] string hasTraining = "", [FromQuery] string programmeId = "")
         {
+            var clearTraining = string.IsNullOrWhiteSpace(clear) == false;
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Controllers/Part1/TrainingController.cs]: I do. I've stopped using ! I find this is clearer with my eyesight.

File: src/Employer/Employer.Web/Controllers/Part1/TrainingController.cs [hunk 3]
```diff
@@ -89,5 +101,65 @@ public IActionResult TrainingHelp(VacancyRouteModel vrm)
...
+
+            var programme = await _orchestrator.GetProgrammeAsync(m.ProgrammeId);
+
+            if(programme == null)
+            {
+                ModelState.AddModelError(nameof(TrainingEditModel.SelectedProgrammeId), InvalidTraining);
+                var vm = await _orchestrator.GetTrainingViewModelAsync(m, user);
+                vm.PageInfo.SetWizard(wizard);
+                return View(vm);
+            }
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Controllers/Part1/TrainingController.cs]: 👍

File: src/Provider/Provider.Web/Controllers/Part1/TrainingController.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Provider/Provider.Web/Controllers/Part1/TrainingController.cs]: :+1:

**Pull request merged**